src/hooks/transactions/useTransactionPreparation.ts

```diff
@@ -33,89 +29,31 @@ interface Props {
   isValidCall?: boolean;
 }
 
-interface FeeAndPayload {
-  payload: any;
-}
-
-export default function useTransactionPreparation({
-  input,
-  type,
-  weightInput,
-  isValidCall = true
-}: Props): FeeAndPayload {
+export default function useTransactionPreparation({ input, type, weightInput, isValidCall = true }: Props) {
   const { areApiReady } = useLoadingApi();
-  const laneId = useLaneId();
+
   const {
-    sourceChainDetails: {
-      apiConnection: { api: sourceApi }
-    },
-    targetChainDetails: { chain: targetChain }
+    sourceChainDetails: { chain: sourceChain }
   } = useSourceTarget();
   const { account } = useAccountContext();
-
-  const [payload, setPayload] = useState<null | {}>(null);
   const { call, weight } = useTransactionType({ input, type, weightInput });
 
-  const { dispatchTransaction } = useUpdateTransactionContext();
-  const { estimatedFeeMethodName } = getSubstrateDynamicNames(targetChain);
+  const calculateFee = useEstimateFee();
+  const getPayload = usePayload();
 
   useEffect(() => {
-    const calculateFee = async () => {
-      // Ignoring custom types missed for TS for now.
-      // Need to apply: https://polkadot.js.org/docs/api/start/typescript.user
-      // @ts-ignore
-      const payloadType = sourceApi.registry.createType('OutboundPayload', payload);
-      // @ts-ignore
-      const messageFeeType = sourceApi.registry.createType('MessageFeeData', {
-        lane_id: laneId,
-        payload: payloadType.toHex()
-      });
-
-      const estimatedFeeCall = await sourceApi.rpc.state.call<Codec>(estimatedFeeMethodName, messageFeeType.toHex());
-
-      // @ts-ignore
-      const estimatedFeeType = sourceApi.registry.createType('Option<Balance>', estimatedFeeCall);
-      const estimatedFee = estimatedFeeType.toString();
-
-      dispatchTransaction(TransactionActionCreators.setEstimateFee(estimatedFee));
+    const asyncCalculateFee = async () => {
+      await calculateFee();
```

Comment:
This is not supposed to be an async function. Also why does it calculate fee only once? It seems to me that passing all parameters to this function would make the entire flow much cleaner and easier to understand. We definitely want to re-calculate the fee each time `payload` changes.

Reply:
Yes, you are right. And that's the exact reason why the code is hard to grasp. It's confusing to see a function that takes no parameters at all, but keeps changing every time some other random parameters change. The only way to figure out what will trigger fee calculation is to follow the code path and that's just entering a rabbit hole.

Also again, why is this function `async`? It should be just a regular function, even if it was `async` there is no point in handling it here cause we don't even expect it to return anything nor we handle errors.

Reply:
you are right on both ... also it is not needed to have an async call.

Reply:
Code updated.